snpdb/models/models_variant.py

```diff
@@ -281,14 +266,23 @@ def __str__(self):
         return s
 
     @staticmethod
-    def from_clean_str(clean_str: str):
-        if full_match := VARIANT_PATTERN.fullmatch(clean_str):
+    def from_string(variant_string: str, regex_pattern=VARIANT_PATTERN):
+        # This will only ever match standard (non-symbolic variants)
+        if full_match := regex_pattern.fullmatch(variant_string):
```

Comment:
Not that it did it before, but should this thow an exception if the string doesn't match the pattern?
Most of the code that calls VariantCoordinate.from_string() expects that there will be a value returned and if not, the code will generally do undesirable things like put Nones into lists etc.

Reply:
yep, better safe than sorry - added simple type checking (not sure what else you want - or whether we can do this automatically)